packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2633,8 +2656,29 @@ class InputDecoration {
   /// the input has focus or text has been entered.
   ///
   /// Defaults to true.
+  ///
+  @Deprecated(
+    'Use floatLabelBehaviour instead. '
+    'This feature was deprecated after v1.13.2.'
+  )
   final bool hasFloatingPlaceholder;
 
+  /// Defines how the floating label should be displayed.
+  ///
+  /// When [FloatLabelBehavior.auto] the label will float to the top only when
+  /// the field is focused or has some text content, otherwise it will appear
+  /// in the field in place of the content.
+  ///
+  /// When [FloatLabelBehavior.always] the label will always float at the top
+  /// of the field above the content.
+  ///
+  /// When [FloatLabelBehavior.never] the label will always will appear
+  /// in the field in place of the content.
```

Comment:
Typo: "will always will"

Also I'm worried that this might be slightly misleading.  It sort of sounds like the label will totally replace the content and the content won't show up at all.  Maybe say this?: "When [FloatLabelBehavior.never] the label will always appear in an empty field in place of the content."